core/arch/arm/sm/sm_a32.S

```diff
@@ -268,6 +268,33 @@ UNWIND(	.fnstart)
 	cps	#CPSR_MODE_MON
 	/* Point just beyond sm_ctx.sec */
 	sub	sp, r0, #(SM_CTX_SIZE - SM_CTX_NSEC)
+
+#ifdef CFG_INIT_CNTVOFF
+	read_scr r0
+	orr	r0, r0, #SCR_NS /* Set NS bit in SCR */
+	write_scr r0
+
+	/*
+	 * Accessing CNTVOFF:
+	 * If the implementation includes the Virtualization Extensions
+	 * this is a RW register, accessible from Hyp mode, and
+	 * from Monitor mode when SCR.NS is set to 1.
+	 * If the implementation includes the Security Extensions
+	 * but not the Virtualization Extensions, an MCRR or MRRC to
+	 * the CNTVOFF encoding is UNPREDICTABLE if executed in Monitor
+	 * mode, regardless of the value of SCR.NS.
+	 * So test VIRT and GENTIMER Mask both here.
+	 */
+	read_idpfr1 r2
+	ands	r2, r2, #(IDPFR1_GENTIMER_MASK | IDPFR1_VIRT_MASK)
```

Comment:
You need both ID_PFR1[generic timer] != 0 and ID_PFR1[virtualization extensions] != 0 to access the register, no?
The proposed test requires only 1 of them to be non null to init the register.

Reply:
fix in next version.